src/Swarm/Game/Step.hs

```diff
@@ -877,8 +878,12 @@ execConst c vs s k = do
         rDir <- use robotOrientation
 
         let nextLoc = loc ^+^ applyTurn d (rDir ? V2 0 0)
-        em <- use entityMap
-        drill <- lookupEntityName "drill" em `isJustOr` Fatal "Drill does not exist?!"
+
+        let toyDrill = lookupByName "drill" ins
+            metalDrill = lookupByName "metal drill" ins
+            insDrill = listToMaybe $ metalDrill <> toyDrill
+
+        drill <- insDrill `isJustOr` Fatal "Drill is required but not installed?!"
```

Comment:
This should not be a `Fatal` error.  The  `Fatal` error from before was for the situation that looking up `drill` in *the entity map* failed, i.e. the system as a whole did not know about the `drill` entity at all.  But the new code is just checking what kind of drill the robot has installed.  It could certainly happen that the robot is trying to execute a `drill` command even though it has no drill installed.  That is a runtime error but not a `Fatal` one.

Reply:
I suspected as much, but what example would escape our capability checking ahead of drilling?

Reply:
Hmm, good question.  I guess you're right, if this happens it means something has gone wrong with capability checking, so perhaps that should actually be a `Fatal` error.

Reply:
Given that even build inside build checks ahaed, I am going to leave it as Fatal and see if it comes up.

```
> build {r <- build {move; drill down; log ""}; log ""};
<r3> : robot
> salvage
```
```
[busy_williamson] build: You do not have required devices:
  - metal drill or drill
  - logger
```